soc/mediatek/mt8xxx/Kconfig.defconfig

```diff
@@ -23,11 +23,22 @@ config MAX_IRQ_PER_AGGREGATOR
 config 2ND_LVL_ISR_TBL_OFFSET
 	default 32
 
+# The 8186/8188 core has only one software interrupt that lives at
+# level 2, underneath other hardware interrupts like timer, so it
+# can't reliably do this.  Unselect so the tests don't try to exercise
+# it.
+config IRQ_OFFLOAD_NESTED
+	default n if SOC_SERIES_MT818X
+	default y
+
 config MTK_ADSP_TIMER
 	default y
 config XTENSA_TIMER
 	default n
 
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 13000000
```

Comment:
awaiting comment from Gerard

Reply:
see e.g.

https://github.com/zephyrproject-rtos/zephyr/blob/b7b4de8afeb93b480b6a8e3bef18c76ea6a82bc9/soc/gd/gd32/gd32f4xx/Kconfig.defconfig.gd32f470#L6-L7

Reply:
Alas, this doesn't seem to work.  This is what I have right now, and it evaluates to a zero instead of 13M.   Variants like specifying the path explicitly (which is a non-starter as the node contains a reg address which we obviously don't want to be copying into kconfig), using an dt_node_array_prop_int with an index of zero, etc... Any guidance?  I actually don't see any other drivers in the tree pulling an integer out of a nodelabel like this.  The `cpus@0` idiom is common, but this isn't a CPU frequency, it's owned by the timer device.

```diff
diff --git a/boards/mediatek/mt8196/mt8196_adsp.dts b/boards/mediatek/mt8196/mt8196_adsp.dts
index 1cc43225f70..e4917d72152 100644
--- a/boards/mediatek/mt8196/mt8196_adsp.dts
+++ b/boards/mediatek/mt8196/mt8196_adsp.dts
@@ -79,6 +79,7 @@ soc {
        ostimer64: ostimer64@1a00b080 {
                compatible = "mediatek,ostimer64";
                reg = <0x1a00b080 28>;
+               clock-frequency = <13000000>;
        };
 
        ostimer0: ostimer@1a00b000 {
diff --git a/soc/mediatek/mt8xxx/Kconfig.defconfig b/soc/mediatek/mt8xxx/Kconfig.defconfig
index b2796b8889d..1f28b2317e5 100644
--- a/soc/mediatek/mt8xxx/Kconfig.defconfig
+++ b/soc/mediatek/mt8xxx/Kconfig.defconfig
@@ -37,7 +37,7 @@ config XTENSA_TIMER
        default n
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
-       default 13000000
+       default $(dt_node_int_prop_int,$(dt_nodelabel_path,ostimer64),clock-frequency)
 
 config MAIN_STACK_SIZE
        default 2048
```